src/iface-ethernet-ip.c

```diff
@@ -89,6 +89,36 @@ static int verify_tag_name_created(struct knot_data_item *data_item)
 		if (!return_aux) {
 			data_item->tag =
 				data_item_aux->tag;
+			strcpy(data_item->string_tag_path,
+			       (const char *)data_item_aux->string_tag_path);
+			rc = 0;
+			break;
+		}
+	}
+
+	return rc;
+}
+
+static int put_tag_name_created(struct knot_data_item *data_item)
+{
+	struct knot_data_item *data_item_aux;
+	int i;
+	int rc = -1;
+
+	for (i = 0; i < thing_ethernet_ip.number_sensor; i++) {
+		int return_aux;
+
+		data_item_aux = l_hashmap_lookup(thing_ethernet_ip.data_items,
+						 L_INT_TO_PTR(i));
+		if (!data_item_aux)
+			rc = -EINVAL;
+
+		return_aux = strcmp(
+			data_item_aux->tag_name,
+			data_item->tag_name);
+		if (!return_aux) {
```

Comment:
(Non-blocking) IMO it would be better if the strcmp function was inside the "if" verification.

```suggestion
		if (!strcmp(data_item_aux->tag_name, data_item->tag_name)) {
```